### DISCUSSION TITLE
Ikea Remote controller sucessfully bound to group on firmware version 24.4.5

### DISCUSSION THREAD

**Discussion by ghost** [Koenkk/zigbee2mqtt#20625] [Devices]
Hello!

Lots of places in the documentation mention that Ikea remote controllers cannot be bound to group, only to individual devices, depending on which version of the firmware they are running.
This has some limitation as most of the remotes can only be bound to a maximum of 7 devices.

However, I managed to bind one of my remote to a group!
Remote is a [E1524/E1810](https://www.zigbee2mqtt.io/devices/E1524_E1810.html) (the one with 5 buttons), and it's running firmware version 24.4.5.

The remote is currently bound to the following clusters, but note that I haven't done any work to try to trace what is the minimal set needed: genBasic, Groups, LevelCtrl, OnOff, genPollCtrl, Scenes.

Just wanted to let people know that it appears to be possible, and if more people can try & confirm that this is indeed now working, we could update the docs!

Cheers!

**Comment by Koenkk** [comment]
Could you make a PR to update the docs? To do this scroll down to the docs page and press "Help to make.."

**Reply by ghost** [reply]
Done: https://github.com/Koenkk/zigbee2mqtt.io/pull/2476

**Comment by sjorge** [comment]
Minimal set is onOff (for just toggel button/or on/off on the onoff remote), you can add levelCtrl for brightness control.

**Comment by kjw1985** [comment]
> However, I managed to bind one of my remote to a group!

Did you do anything special? I tried this today and could bind my remote to a group (via GUI or mqtt). However the buttons don't seem to do anything. I see the toggle action or brightness action published to mqtt, however nothing happens to my light group. 
However if I bind it to a light it seems at least to toggle the light on/off.
I have 24.4.5 installed on the remote and z2m 1.35.3.

**Reply by ghost** [reply]
I think you need to unbind to the coordinator for it to work with a group.
I only have the binding between the source endpoint 1 and the group on mine.

**Reply by kjw1985** [reply]
Thank you for your quick answer. I will try this when I have the time and comment back if this works. Also need to look into the need of the binding with the coordinator.

**Reply by danmichaelo** [reply]
Thanks a lot! Had the same problem and unbinding from the coordinator helped!

**Comment by cipisek9** [comment]
Hi, do you have any tips on how to configure it? I’ve tried to bind a group of IKEA bulbs, and everything seems to be okay, but when I press the button, nothing happens. I have firmware version 24.4.5, and according to the [website](https://www.zigbee2mqtt.io/devices/E1524_E1810.html), it should support binding to both individual devices and groups. Thank you.

**Reply by kjw1985** [reply]
I tried it today as for some reason 1 of my 5 bindend lamps did not respond anymore. But whatever I tried groups didn't seem to work. I could bind the group (used on off and lvlcobtrol), but no message was send after a press. Reconfigure would also not work (as stated on the page). So binded all devices separately again.

**Reply by ghost** [reply]
What type of bulb do you have? Maybe there are different firmware for them and they do not support group binding :(
Mine are LED1536G5 (fw version 2.3.087), LED2101G4 (fw version 1.1.003), LED1622G12 (fw version 2.3.094), LED1649C5 (fw version 2.3.094), all grouped and bound to a E1524/E1810 (fw version 24.4.5) remote controller.

On the remote configuration page in z2mqtt, source endpoint is 1, destination is the group and clusters are as follows: genBasic, Groups, LevelCtrl, OnOff, genPollCtrl, Scenes

The remote was unbound from the controller (thus it does not transmit its input to z2mqtt).

**Reply by kjw1985** [reply]
Mine are 5 times the LED1537R6/LED1739R5  with fw 2.3.087.  I did remove unbind the controller indeed. I forgot that it won't register the presses anymore. Maybe the the selected clusters does matter?

**Comment by sbrunk** [comment]
First of all, thanks @chemicalstorm, I got group binding working too with a [E1524/E1810](https://www.zigbee2mqtt.io/devices/E1524_E1810.html), thanks to your description above, and especially this:
> I think you need to unbind to the coordinator for it to work with a group.

I have a also have a [E2001/E2002 / STYRBAR](https://www.zigbee2mqtt.io/devices/E2001_E2002.html), and according to the [current docs](https://www.zigbee2mqtt.io/devices/E2001_E2002.html), or https://github.com/Koenkk/zigbee2mqtt/issues/17238, it doesn't do group binding with newer firmware versions. So I got curious and tried your approach, and got it working too!

I played a little bit with it, and the important thing seems to be not to bind any clusters to **both**, remote **and** coordinator, in which case the lights in the group don't react to commands from the remote corresponding to that cluster any more. It seems like non-overlapping clusters are fine, with i.e. the setup shown in the screenshot the remote is working fine & the remote commands also still arrive as MQTT messages (haven't checked yet if Ota still works this way).

![grafik](https://github.com/user-attachments/assets/8b76ae0b-4139-4fb0-8784-f8f5f128d710)

@Koenkk would you like me to draft a PR for the docs for this? Might be helpful to others as well. I've seen quite a few issues/discussions about it.

**Reply by Koenkk** [reply]
> @Koenkk would you like me to draft a PR for the docs for this? Might be helpful to others as well. I've seen quite a few issues/discussions about it.

yes please!

**Reply by SteinVanLoon** [reply]
How did you succeed in only binding to the coordinator for Ota and nothing else?

**Reply by sbrunk** [reply]
1. bind the clusters to the remote (as in the 1. screenshot)
2. select everything except Ota on the coordinator binding and click unbind

Don't forget to click a button on the remote before doing this to wake it up.

**Reply by esolitos** [reply]
@sbrunk What firmware are yo running on the E2001/E2002? I've bought new ones and they came with 2.4.17.
I tried the suggested method with no avail. It seems that I am able to  whatever I do I end up in a situation where the Bind/Unbind options completely stop working and I need to Reset the device, remove it from the network and make it join again.

Here is an example (after many many retries), in this picture shows that I was able to:
- Add binding for genOnOff on the "Dining" group
- Remove binding of genOnOff to theCoordinator
- 💥 Can't bind/unbind anymore.. 
<img width="2041" height="437" alt="image" src="https://github.com/user-attachments/assets/6b587bee-1f02-4fa8-8577-a51af1e4eabb" />

This is only an example.. I'll try to understand exactly what is the sequence to reproduce, but I might end up returning the 8 Styrbar that I bought otherwise.. 😢😢

**Comment by geezer78** [comment]
I too have been trying to bind my 5 button remote to a group and found this thread.
however I cannot seem to bind it to my group.
I was thinking it might be due to the fact that my group is too large (10) so tried with a smaller one (4) but was unable to bind to just the group.
Latest firmware for both button and lights.
I have tried multiple times and did notice some times it seemed to bins successfully but then nothing happens. It did the show multiple messages as if binding to each light individually as well...

It would be very appreciated if we can fix this.
the same button and light could easily be bound to each other with the ikea hub.

**Comment by datbilling** [comment]
I’ve tried the steps mentioned but still cannot get the bulbs in the group to respond to the E1524 Tradfri remote (firmware 24.4.5).

It only works when I bind the remote directly to each bulb, which isn’t a solution as 1 of the 4 bulbs often doesn’t respond and gets out of sync with the rest.

Does anyone have any more tips to get this working please?

**Comment by DominikMa** [comment]
After fighting with the Tradfri remote for hours, I finally got them working with groups.
Key was to follow exactly [sbrunks tips](https://github.com/Koenkk/zigbee2mqtt/discussions/20625#discussioncomment-11987769). Any deviation from the following would lead, at least for me, to the remote not be bound to the group or the group not responding.

The following worked for me:
1. After pairing the remote bind genBasic, genIdentify, genOta to the coordinator. Only keep the three in the Cluster field, to not try to rebind the other clusters. Keep pressing any button every few seconds. After that all "gen..." clusters should be bound to the coordinator.
2. Bind ALL "gen..." clusters besides "genOta" to the group. Again pressing a button while binding.
3. In the Cluster field for the coordinator remove only the "genOta" cluster to unbind all but the genOta cluster from the coordinator. Keep pressing buttons.
4. If everything got right, the coordinator has only the genOta cluster and the group all other. When you press any button on the remote now it still should appear in the zigbee2mqtt logs. If not the group will not be reacting.

This worked for me with one remote with firmware 2.4.17, one with 2.4.16 and one with 2.4.5.
Hope this helps anybody.

**Reply by datbilling** [reply]
It finally works! Thank you so much for sharing this! 

I couldn't get the group binding to work until I followed your instructions.

**Reply by esolitos** [reply]
Ha! I can confirm, following the steps, WITHOUT ANY DEVIATION seems to actually work!


Now I need to figure out why 1 of the STYRBAR when connected **only** shows up with `genPollCtrl` bind to Coordinator..

**Comment by andrei-lazarov** [comment]
Thanks everyone for figuring group binding! 😁 
I did some testing on STYRBAR and SYMFONISK. I will update the docs accordingly:

**The minimal instructions for group binding should be:**
1. Bind `genBasic` to group
2. Unbind _everything_ else, including coordinator
(needed on newer fw, not needed on older fw)

This enables all actions: toggle, dim, scenes. And the coordinator still receives the commands.

- Older fw can bind to multiple devices + multiple groups
- Newer fw can bind to multiple devices **or** to one group

Binding to individual devices depends on the fw versions:
- on older fw, you need to bind a single (any) cluster, e.g. `genOnOff` 
  - it enables all actions
  - if you bind multiple clusters, the remote may repeat the commands!!
- on newer fw, you need to bind every cluster you need: `genOnOff`, `genLevelCtrl`, `genScenes`
  - each enables its respective action

**Comment by culdaraul** [comment]
Unfortunately, I was not able to use the group binding anymore with 2.4.16 fw. At some point, one of the remotes (out of 5) was not even able to bind itself correctly to the coordinator. All my issues were gone once I performed the downgrade to 2.4.11, everything works again as expected.
